build_defs/go.build_defs

```diff
@@ -58,6 +58,9 @@ def go_toolchain(name:str, url:str|dict = '', version:str = '', hashes:list = []
     for arch in architectures:
         goos, _, goarch = arch.partition("_")
         cmd += f' && (export GOOS={goos} && export GOARCH={goarch} && $OUT/bin/go install{tag_flag} --trimpath std)'
+    # TODO(jpoole): we should parse the version and figure out if we're post 1.20 but asp doesn't really facilitate that
```

Comment:
could just do `if '1.2' in version` - it's not any kind of true parse but would work fine, and we can get rid of it long before 1.30 comes along.

Reply:
Apologies for hijacking the review but ASP *does* facilitate that (according to the docs and [ASP's source code itself](https://github.com/thought-machine/please/blob/5b08a9a4c8b1b73453e51ad8d35906d01be1b8dd/src/parse/asp/builtins.go#L1213-L1229)). Something like this should work.

```python
if semver_check(version, ">= 1.20.0, < 1.30.0"):
```

Reply:
:confused: when did we implement that? 

Thanks for pointing that out!